src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullChecking.cs

```diff
@@ -49,7 +50,18 @@ private BoundBlock RewriteNullChecking(BoundBlock block)
 
         }
 
-        private static BoundStatement ConstructIfStatementForParameter(SourceParameterSymbolBase parameter, SyntheticBoundNodeFactory factory)
+        private static bool GetParameterIsNullChecked(ParameterSymbol x)
```

Comment:
Should `IsNullChecked` be defined on `ParameterSymbol` rather than just derived classes?

Reply:
It opens up a whole other set of child classes like TupleParameterSymbol I'd have to implement as well. Should I?

Reply:
There shouldn't be more than 6 or so implementations.

---
In reply to: [300920432](https://github.com/dotnet/roslyn/pull/36979#discussion_r300920432) [](ancestors = 300920432)

Reply:
Will most implementations just be `return false`? If so does it perhaps make sense to include this implementation on `ParameterSymbol` and make the method virtual for `SourceParameterSymbolBase` and `SynthesizedParameterSymbolBase` to override?

---
In reply to: [301133543](https://github.com/dotnet/roslyn/pull/36979#discussion_r301133543) [](ancestors = 301133543,300920432)

Reply:
I don't have a strong preference either way on this :)

---
In reply to: [301208576](https://github.com/dotnet/roslyn/pull/36979#discussion_r301208576) [](ancestors = 301208576,301133543,300920432)